code/modules/client/preferences.dm

```diff
@@ -1148,12 +1146,18 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 
 				if("name")
 					var/new_name = input(user, "Choose your character's name:", "Character Preference")  as text|null
-					if(new_name)
-						new_name = reject_bad_name(new_name)
+					if(pref_species.id != "ipc")

```

Comment:
Having something like species/proc/check_valid_name() instead of clustering this all here

Reply:
Not entirely sure how to do this